### PR TITLE
magento/magento2#1839 customer addresses export missing customer_id #44684

### PR DESCRIPTION
**Pull request by keyurshah070**
### Fixed Issues (if relevant)

1. magento/magento2#1839: customer addresses export missing customer_id
2. Customer Id was missing on customer address export
3. Added customer_id column

### Manual testing scenarios

1. Export with some dummy customer and check for the column

Also attaching screenshot of excel sheet with customer_id

customer_address_export

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by keyurshah070** [comment]
Any update for merging this branch? @mzeis

**Comment by mzeis** [comment]
@keyurshah070 Sorry for letting you wait! I didn't get around to merge it yet but I'll review as soon as I can.

**Comment by keyurshah070** [comment]
Thanks for the updates and no need to be sorry :) @mzeis

**Review by mzeis — APPROVED** [review]

File: app/code/Magento/CustomerImportExport/Model/Export/Address.php [hunk 1]
```diff
@@ -27,6 +27,8 @@ class Address extends \Magento\ImportExport\Model\Export\Entity\AbstractEav
 
     const COLUMN_ADDRESS_ID = '_entity_id';
 
+    const COLUMN_CUSTOMER_ID = '_customer_id';
```
Comment (mzeis) [inline app/code/Magento/CustomerImportExport/Model/Export/Address.php:30]: Adding a constant is a backwards incompatible change (see http://devdocs.magento.com/guides/v2.1/ext-best-practices/extension-coding/backwards-compatible-development/index.html#documentation-of-backwards-incompatible-changes).

Suggestion by the core team:
* in this PR for `2.2-develop`, don't use a constant but hard-code the string
* in addition, create a PR for `2.3-develop` that uses the constant.

This way, we have a quicker fix for shops on 2.2 but we use your approach (with is the best practice!) for 2.3

**Review by mzeis — CHANGES_REQUESTED** [review]

**Comment by mzeis** [comment]
@keyurshah070 

Please see the code review for suggested changes to keep the code backwards-compatible.

Also, an integration test in `dev/tests/integration/testsuite/Magento/CustomerImportExport/Model/Export/CustomerTest.php` is failing:

~~~
There was 1 failure:
22-Nov-2017 00:43:05         [exec] 
22-Nov-2017 00:43:05         [exec] 1) Magento\CustomerImportExport\Model\Export\CustomAddressTest::testExport
22-Nov-2017 00:43:05         [exec] Failed asserting that two strings are equal.
22-Nov-2017 00:43:05         [exec] --- Expected
22-Nov-2017 00:43:05         [exec] +++ Actual
22-Nov-2017 00:43:05         [exec] @@ @@
22-Nov-2017 00:43:05         [exec] -'_website,_email,_entity_id,city,company,country_id,fax,firstname,lastname,middlename,multi_select_attribute_code,postcode,prefix,region,region_id,street,suffix,telephone,test_select_code,vat_id,vat_is_valid,vat_request_date,vat_request_id,vat_request_success,_address_default_billing_,_address_default_shipping_\n
22-Nov-2017 00:43:05         [exec] -admin,BetsyParker@example.com,1,Philadelphia,,US,,Betsy,Parker,,"Option 2",19107,,Pennsylvania,51,"1079 Rocky Road",,215-629-9720,Second,,,,,,1,\n
22-Nov-2017 00:43:05         [exec] -admin,BetsyParker@example.com,2,Fayetteville,,US,,Anthony,Nealy,,"Option 1,Option 3",72701,,Arkansas,5,"3176 Cambridge Court",,479-899-9849,First,,,,,,,1\n
22-Nov-2017 00:43:05         [exec] +'_website,_email,_customer_id,_entity_id,city,company,country_id,fax,firstname,lastname,middlename,multi_select_attribute_code,postcode,prefix,region,region_id,street,suffix,telephone,test_select_code,vat_id,vat_is_valid,vat_request_date,vat_request_id,vat_request_success,_address_default_billing_,_address_default_shipping_\n
22-Nov-2017 00:43:05         [exec] +admin,BetsyParker@example.com,1,1,Philadelphia,,US,,Betsy,Parker,,"Option 2",19107,,Pennsylvania,51,"1079 Rocky Road",,215-629-9720,Second,,,,,,1,\n
22-Nov-2017 00:43:05         [exec] +admin,BetsyParker@example.com,1,2,Fayetteville,,US,,Anthony,Nealy,,"Option 1,Option 3",72701,,Arkansas,5,"3176 Cambridge Court",,479-899-9849,First,,,,,,,1\n
22-Nov-2017 00:43:05         [exec]  '
~~~
Please can you fix it?

Thanks for taking the time!

**Comment by mzeis** [comment]
@keyurshah070 Does this mean you don't want to work on this PR anymore? Can we help with something?

**Pull request closed without merging**